organizations/v0/tests/test_views.py

```diff
@@ -56,3 +58,26 @@ def test_nonexistent_organization(self):
         url = reverse('v0:organization-detail', kwargs={'short_name': 'dummy'})
         response = self.client.get(url)
         self.assertEqual(response.status_code, 404)
+
+    def test_oauth2(self):
+        """Verify that the API can handle OAuth 2.0 access tokens."""
+        oauth2_client = Client.objects.create(client_type=CONFIDENTIAL)
+        access_token = AccessToken.objects.create(
+            token='fake-access-token',
+            client=oauth2_client,
+            user=self.user,
+        )
+
+        self.client.logout()
+
+        response = self.client.get(
+            self.organization_list_url,
+            HTTP_AUTHORIZATION='Bearer {}'.format(access_token)
+        )
+        self.assertEqual(response.status_code, 200)
+
+        response = self.client.get(
+            self.organization_list_url,
+            HTTP_AUTHORIZATION='Bearer {}'.format('nonexistent-access-token')
+        )
+        self.assertEqual(response.status_code, 403)
```

Comment:
Turns out re-ordering the classes does work as expected. I was being thrown off by some test cases. The view now returns a 401 when authentication fails.